src/app/shell/workshop-details/workshop-page/workshop-page.component.ts

```diff
@@ -7,41 +7,51 @@ import { Provider } from 'src/app/shared/models/provider.model';
 import { Workshop, WorkshopCard } from 'src/app/shared/models/workshop.model';
 import { AppState } from 'src/app/shared/store/app.state';
 import { environment } from 'src/environments/environment';
-interface imgPath {
-  path: string;
-}
+import { UserState } from 'src/app/shared/store/user.state';
+import { filter, takeUntil } from 'rxjs/operators';
+import { imgPath } from 'src/app/shared/models/carousel.model';
+
 @Component({
   selector: 'app-workshop-page',
   templateUrl: './workshop-page.component.html',
   styleUrls: ['./workshop-page.component.scss']
 })
 
 export class WorkshopPageComponent implements OnInit, OnDestroy {
+  
   readonly Role: typeof Role = Role;
   public categoryIcons = CategoryIcons;
-  @Input() workshop: Workshop;
   @Input() provider: Provider;
   @Input() providerWorkshops: WorkshopCard[];
   @Input() role: string;
-
+  
+  @Select(UserState.selectedWorkshop) workshop$: Observable<Workshop>;
   @Select(AppState.isMobileScreen) isMobileScreen$: Observable<boolean>;
 
   destroy$: Subject<boolean> = new Subject<boolean>();
 
   authServer: string = environment.serverUrl;
   imgUrl = `/api/v1/PublicImage/`;
   images: imgPath[] = [];
+  workshop: Workshop;
 
   constructor() { }
 
   ngOnInit(): void {
-    this.getWorkshopImages();
+    this.workshop$.pipe(
+      filter((workshop: Workshop) => !!workshop),
+      takeUntil(this.destroy$)
+    ).subscribe((workshop: Workshop) => {
+      this.workshop = workshop;
+      this.getWorkshopImages();
+    })
   }
 
   private getWorkshopImages(): void {
     if (this.workshop?.imageIds.length) {
       this.images = this.workshop.imageIds.map((imgId) => { return { path: this.authServer + this.imgUrl + imgId } })
     } else {
+      this.images = [];
       this.images.push({ path: 'assets/images/groupimages/workshop-img.png' })
```

Comment:
semicolon;

 this.images = [{ path: 'assets/images/groupimages/workshop-img.png' }];

Reply:
done